builtin/k8s/platform.go

```diff
@@ -132,15 +132,22 @@ func (p *Platform) Deploy(
 		return nil, err
 	}
 
-	if p.config.ServicePort == 0 {
-		p.config.ServicePort = 3000
+	// Determine service ports for container
+	if len(p.config.ServicePorts) == 0 {
+		p.config.ServicePorts = []uint{3000}
+		if sp := p.config.ServicePort; sp != 0 {
+			p.config.ServicePorts[0] = sp
+		}
+	} else if p.config.ServicePort != 0 {
+		err := fmt.Errorf("Cannot specify both service_port and service_ports")
+		return nil, err
```

Comment:
@briancain I'm trying to pursue the unified configuration here, but seems like `gohcl` doesn't support automatically coercing to a Go-type for an `interface{}` struct field, and the common pattern is to use `cty.Value` for a field that can hold various types.

Such marshaling of `cty.Value` works in my standalone tests, however, it seems like the plugin system somehow loses the struct definition and only carries it as an anonymous struct, and HCL's `DecodeExpression` panics with `no cty.Type for struct`.

Reply:
Hey @rykov - Interesting. I think it would be better to not have `ServicePort` be an empty `interface{}` then, but something gohcl can handle like you say? So instead we define it as an `hcl.Expression`, and then your `coerceConfigPortArray` function might have a better time decoding `ServicePort`? If it helps, this might be pretty similar to something Terraform does for parsing resources: https://github.com/hashicorp/terraform/blob/master/configs/resource.go#L18-L20

I noticed now that if you set it as `interface{}` and `service_port=3000`, it ends up with an unexpected type that I didn't account for from the diff above: a `map[string]interface{}`. But using `hcl.Expression` should get you where we want to be and ultimately feels like the right move for parsing something out of a waypoint.hcl config :smile: 

Let me know if that makes sense or if you need extra help! Sorry this request was a bit much :sweat_smile: :heart: I think the end result UX wise will be worth it :pray: Thanks again for tackling this for us, and thanks for writing tests! :smile:

Reply:
Hi @briancain. Looks like that approach is also not passing correctly through the plugin/protobufs layer.  I'm getting the following error if I define `ServicePort` as `hcl.Expression`:

```
<nil>: rpc error: code = Unknown desc = json: cannot unmarshal object into Go
struct field Config.ServicePort of type hcl.Expression; 
```

Now I'm starting to remember why I implemented a separate `ServicePorts` variable. 😅

Reply:
Hey @rykov ! Interesting, well I've give your PR a shot today or this week and see what I can do and let you know what I find. :smile:

Reply:
Hello again @rykov ! So internally we talked about this, and going forward we think it makes sense to simply update `ServerPort` and other multi-port vars to only be an array. It will end up breaking backwards compatibility, but I think going forward having 1 config value for defining Port is best. And also it looks like due to how the plugin system works, it isn't possible to have the kind of loose variable def that I was hoping we could work with. I'll use these commits as the base to include your contributions, but I'll end up updating the other supported platforms for this too. Thanks again for getting this started! Once I open the PR I'll close this one, but I'll be sure to include the original commits. Thanks!

Reply:
Hi @briancain. I really had my hopes up about making this work. I guess having a single multi-port array value is the best alternative. But if they're going to be compatibility-breaking changes anyway, I'd suggest using plural key names (e.g. `server_ports`, `node_ports`, etc) - makes more sense to me as a user if the expected value is an array. Thanks for following up, and pushing this through. I look forward to having this functionality integrated.